common/src/allocations/monitor.rs

```diff
@@ -135,64 +45,93 @@ pub fn indexer_allocations(
+}
+
 #[cfg(test)]
 mod test {
-    use wiremock::{
-        matchers::{body_string_contains, method, path},
-        Mock, MockServer, ResponseTemplate,
-    };
+    const NETWORK_SUBGRAPH_URL: &str =
+        "https://api.thegraph.com/subgraphs/name/graphprotocol/graph-network-arbitrum";
+    use std::str::FromStr;
 
-    use crate::{prelude::SubgraphClient, subgraph_client::DeploymentDetails, test_vectors};
+    use crate::{prelude::SubgraphClient, subgraph_client::DeploymentDetails};
 
     use super::*;
 
-    async fn setup_mock_network_subgraph() -> (&'static SubgraphClient, MockServer) {
-        // Set up a mock network subgraph
-        let mock_server = MockServer::start().await;
-        let network_subgraph = SubgraphClient::new(
+    fn network_subgraph_client() -> &'static SubgraphClient {
+        Box::leak(Box::new(SubgraphClient::new(
             reqwest::Client::new(),
             None,
-            DeploymentDetails::for_query_url(&format!(
-                "{}/subgraphs/id/{}",
-                &mock_server.uri(),
-                *test_vectors::NETWORK_SUBGRAPH_DEPLOYMENT
-            ))
-            .unwrap(),
-        );
-
-        // Mock result for allocations query
-        mock_server
-            .register(
-                Mock::given(method("POST"))
-                    .and(path(format!(
-                        "/subgraphs/id/{}",
-                        *test_vectors::NETWORK_SUBGRAPH_DEPLOYMENT
-                    )))
-                    .and(body_string_contains("activeAllocations"))
-                    .respond_with(ResponseTemplate::new(200).set_body_raw(
-                        test_vectors::ALLOCATIONS_QUERY_RESPONSE,
-                        "application/json",
-                    )),
-            )
-            .await;
-
-        (Box::leak(Box::new(network_subgraph)), mock_server)
+            DeploymentDetails::for_query_url(NETWORK_SUBGRAPH_URL).unwrap(),
+        )))
     }
 
-    #[test_log::test(tokio::test)]
-    async fn test_parses_allocation_data_from_network_subgraph_correctly() {
-        let (network_subgraph, _mock_server) = setup_mock_network_subgraph().await;
-
-        let allocations = indexer_allocations(
-            network_subgraph,
-            *test_vectors::INDEXER_ADDRESS,
-            Duration::from_secs(60),
-            Duration::from_secs(300),
-        );
-
-        assert_eq!(
-            allocations.value().await.unwrap(),
-            *test_vectors::INDEXER_ALLOCATIONS
-        );
+    #[tokio::test]
+    async fn test_network_query() {
+        let result = get_allocations(
+            network_subgraph_client(),
+            Address::from_str("0x326c584e0f0eab1f1f83c93cc6ae1acc0feba0bc").unwrap(),
+            Duration::from_secs(1712448507),
+        )
+        .await;
+        match result {
+            Ok(allocations) => {
+                assert!(allocations.len() > 1)
+            }
+            Err(e) => panic!("Error: {:#?}", e),
+        }
```

Comment:
And replace `1` with the min number of allocations we're expecting to get from that indexer.
```suggestion
        assert!(result.unwrap().len() > 1)
```

Reply:
It has more than 2000, 2049 actually, but I dont think we can just "hardcode" a value, cause this number will increase and also there will be a point where the closedAllocations will be off from the threshold, But i do believe setting a number like more than 2000 should be okay for this case (already tried it out)

Reply:
Sure, it can only increase anyway, so it's good

Reply:
I see [812965b](https://github.com/graphprotocol/indexer-rs/pull/178/commits/812965ba829248e7f8bc4a9a5279ddb02428e540) but how about the code refactor that I suggested?

Reply:
Could you point me to it again, I rebased with main since there where some conflicts and cant see it no more